pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -192,143 +192,102 @@ public static CompletableFuture<ByteBuf> lookupTopicAsync(PulsarService pulsarSe
                 }
-                validationFuture.complete(newLookupResponse(differentClusterData.getBrokerServiceUrl(),
-                        differentClusterData.getBrokerServiceUrlTls(), true, LookupType.Redirect, requestId, false));
-            } else {
                 // (2) authorize client
-                try {
-                    checkAuthorization(pulsarService, topicName, clientAppId, authenticationData);
-                } catch (RestException authException) {
-                    log.warn("Failed to authorized {} on cluster {}", clientAppId, topicName.toString());
-                    validationFuture.complete(newLookupErrorResponse(ServerError.AuthorizationError,
-                            authException.getMessage(), requestId));
-                    return;
-                } catch (Exception e) {
-                    log.warn("Unknown error while authorizing {} on cluster {}", clientAppId, topicName.toString());
-                    validationFuture.completeExceptionally(e);
-                    return;
+                return validateTopicOperationAsync(pulsarService, topicName,
+                        clientAppId, TopicOperation.LOOKUP, authenticationData)
+                        // (3) validate global namespace
+                        .thenCompose(__ ->
+                            checkLocalOrGetPeerReplicationCluster(pulsarService, topicName.getNamespaceObject())
+                                .thenApply(peerClusterData -> {
+                                    if (peerClusterData != null) {
+                                        // if peer-cluster-data is present it means namespace is owned by
+                                        // that peer-cluster and request should be redirect to the peer-cluster
+                                        if (StringUtils.isBlank(peerClusterData.getBrokerServiceUrl())
+                                                && StringUtils.isBlank(peerClusterData.getBrokerServiceUrlTls())) {
+                                            return Optional.of(newLookupErrorResponse(ServerError.MetadataError,
+                                                        "Redirected cluster's brokerService url is not configured",
+                                                            requestId));
+                                        }
+                                        return Optional.of(newLookupResponse(peerClusterData.getBrokerServiceUrl(),
+                                                peerClusterData.getBrokerServiceUrlTls(), true,
+                                                        LookupType.Redirect, requestId, false));
+                                    }
+                                    // (4) all validation passed: initiate lookup
+                                    return Optional.empty();
+                                }).exceptionally(ex ->
+                                        Optional.of(newLookupErrorResponse(ServerError.MetadataError,
```

Comment:
need a return statement？

Reply:
This is a lambda expression in a single line, we don't need to write a redundant return.
I'm not sure if I missing something, please let me know what you think. thanks!

Reply:
LGTM